### PR TITLE
Fix generic accessor semantics tests

### PR DESCRIPTION
**Pull request by aleksmesh**
Moved semantics checks for accessor with target == host_task inside the handler.host_task command.

### CODE REVIEW

File: tests/accessor/generic_accessor_semantics.cpp [hunk 1]
```diff
@@ -79,48 +79,100 @@ DISABLED_FOR_TEST_CASE(hipSYCL)
...
+          .submit([&](sycl::handler& cgh) {
+            auto acc_result = buffer_result.get_access(cgh);
+            sycl::accessor<int, 1, sycl::access_mode::read_write,
+                           sycl::target::host_task>
+                t0{buffer, cgh};
+            cgh.host_task([=] {
+              t0[0] = val;
+              sycl::accessor<int, 1, sycl::access_mode::read_write,
+                             sycl::target::host_task>
+                  t1(t0);
```
Comment (keryell) [inline tests/accessor/generic_accessor_semantics.cpp:122]:
```suggestion
                  t1{t0};
```
to be coherent with line 117?

File: tests/accessor/generic_accessor_semantics.cpp [hunk 2]
```diff
@@ -79,48 +79,100 @@ DISABLED_FOR_TEST_CASE(hipSYCL)
...
```
Comment (keryell) [inline tests/accessor/generic_accessor_semantics.cpp:145]:
```suggestion
                  t1{t0};
```
to be coherent with line 140?

File: tests/accessor/generic_accessor_semantics.cpp [hunk 3]
```diff
@@ -79,48 +79,100 @@ DISABLED_FOR_TEST_CASE(hipSYCL)
...
+          .submit([&](sycl::handler& cgh) {
+            auto acc_result = buffer_result.get_access(cgh);
+            sycl::accessor<int, 1, sycl::access_mode::read_write,
+                           sycl::target::host_task>
+                t0{buffer, cgh};
+            cgh.host_task([=] {
+              t0[0] = val;
+              const sycl::accessor<int, 1, sycl::access_mode::read_write,
+                                   sycl::target::host_task>
+                  t1(t0);
```
Comment (keryell) [inline tests/accessor/generic_accessor_semantics.cpp:168]:
```suggestion
                  t1{t0};
```
and so on everywhere?

Comment (keryell) [inline tests/accessor/generic_accessor_semantics.cpp:168]:
```suggestion
                  t1{t0};
```
and so on everywhere?

**Review by keryell — APPROVED** [review]
Thanks.

**Comment by tomdeakin** [comment]
Please resolve suggestions, then ready to merge.

File: tests/accessor/generic_accessor_semantics.cpp (hunk 1 above)
Comment (aleksmesh) [reply tests/accessor/generic_accessor_semantics.cpp]: Done. Thanks for suggestion.

File: tests/accessor/generic_accessor_semantics.cpp (hunk 2 above)
Comment (aleksmesh) [reply tests/accessor/generic_accessor_semantics.cpp]: Done.

File: tests/accessor/generic_accessor_semantics.cpp (hunk 3 above)
Comment (aleksmesh) [reply tests/accessor/generic_accessor_semantics.cpp]: Fixed everywhere.

Comment (aleksmesh) [reply tests/accessor/generic_accessor_semantics.cpp]: Done. Thanks!

**Pull request merged**